src/models/tools/geometry/jxg-comment.ts

```diff
@@ -69,6 +70,7 @@ export const commentChangeAgent: JXGChangeAgent = {
   create: (board, change) => {
     const changeProps: any = change.properties || {};
     const commentProps = {
+      id: uniqueId(),
```

Comment:
It seems like these IDs get generated both in `geometry-content.ts` and here in the change handlers - what are the best practices here?

Reply:
The way I have been thinking about it is that 1) it's better for us to generate IDs than to let JSXGraph do so; 2) it's better for those IDs to be in the content, so clients that build change objects should include IDs; 3) the jxg- change agents should not assume that geometry-content.ts is their only client, so they should apply the ID if the client didn't.

Now that you ask the question, however, I realize that a counter-argument could be made that clients that want to set the ID should include it in the change object and that clients that don't set the ID in the change object can be assumed to either not care about the ID or perhaps even prefer to have JSXGraph assign the ID. In either case, perhaps it isn't necessary/useful for the jxg- change agents to assign an ID. For now, however, the jxg- change agents generally are applying IDs, so there's a consistency argument to be made as well, although it probably doesn't matter since geometry-content.ts generally applies IDs anyway.